src/command_modules/azure-cli-botservice/azure/cli/command_modules/botservice/_params.py

```diff
@@ -0,0 +1,130 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+from knack.arguments import CLIArgumentType
+
+from azure.cli.core.commands.parameters import (
+    resource_group_name_type,
+    get_enum_type,
+    get_three_state_flag)
+
+name_arg_type = CLIArgumentType(metavar='NAME', configured_default='botname', id_part='Name')
+
+
+# pylint: disable=line-too-long,too-many-statements
+def load_arguments(self, _):
+    with self.argument_context('bot') as c:
+        c.argument('resource_group_name', arg_type=resource_group_name_type)
+        c.argument('resource_name', options_list=['--name', '-n'], help='the Resource Name of the bot.', arg_type=name_arg_type)
+
+    with self.argument_context('bot create') as c:
+        c.argument('sku_name', options_list=['--sku'], arg_type=get_enum_type(['F0', 'S1']), help='the Sku of the Bot', arg_group='Registration Bot Specific')
+        c.argument('kind', options_list=['--kind', '-k'], arg_type=get_enum_type(['registration', 'function', 'webapp']), help='The Kind of the Bot.')
+        c.argument('display_name', help='the Display Name of the bot.If not specified, defaults to the name of the bot.', arg_group='Registration Bot Specific')
+        c.argument('description', options_list=['--description', '-d'], help='the Description of the bot.', arg_group='Registration Bot Specific')
+        c.argument('endpoint', options_list=['-e', '--endpoint'], help='the Messaging Endpoint of the bot.', arg_group='Registration Bot Specific')
+        c.argument('msa_app_id', options_list=['--appid'], help='the msa account id to be used with the bot.')
+        c.argument('password', options_list=['-p', '--password'], help='the msa password for the bot from developer portal.')
+        c.argument('storageAccountName', options_list=['-s', '--storage'], help='Storage Account Name to be used with the bot.If one is not provided, a new account will be created.', arg_group='Web/Function Bot Specific')
+        c.argument('tags', help='set of tags to add to the bot.')
```

Comment:
Have you tried setting tags?  This does not apply the "tags_type" from core, so I would expect that this doesn't work.

Reply:
yes, i had missed this because i wasnt testing for it. I have fixed the tags implementation now and have added a test to ensure that it works